ykrt/src/compile/jitc_yk/jit_ir/mod.rs

```diff
@@ -3187,6 +3271,21 @@ impl GuardInst {
     }
 }
 
+#[derive(Debug)]
+pub(crate) enum HasGuardInfo {
```

Comment:
This needs a quick doc string (and, ideally, a different name, but that's more subjective).

Reply:
Looking at this I realise that other things also need renaming now. For example the `guard_to_deopt` function in the codegen should probably we called something like `push_deopt` and `self.guards` should become `self.deopts`. It might be better to do that in a quick follow up PR once this goes in. As for renaming `HasGuardInfo` my mind is pulling a blank at the moment. The best I could come up with is `DeoptProvider`.

To be honest this would have been better as a trait, but I couldn't get it through the borrow checker, so an enum was much simpler.

Reply:
I'm pretty sure this commit makes things worse I'm afraid :/ Most of these really should be "guard", not "deopt", IMHO.

Reply:
In retrospect, I suggest we revert https://github.com/ykjit/yk/pull/1808/commits/7c56427d256f09ff710713b6577fce3bcf59aab9. We can merge this PR without it IMHO.